File: Recursion/PowerOf.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        
        System.out.println("Enter a base: ");
        int b = in.nextInt();
        System.out.println("Enter a power: ");
        int p = in.nextInt();
        
        int result = power(b,p);
        System.out.printf("%d^%d = %d",b,p,result);
    }
    public static int power(int b, int p){
        if(p == 1){return b;}
        
        int t = power(b,p/2) * power(b,p/2);
        
        if(p % 2 == 0){
            return t;
        }
        return t * b;
    }
}

File: Recursion/PowerOf.rs
fn main(){
    let b: i16 = 10;
    let p: i16 = 2;
    let result = powerof(b,p);
    
    println!("{}^{} = {}",b,p,result);
}
fn powerof(b: i16, p: i16) -> i16{
    if p == 0 { return 1;}
    
    let t = powerof(b,p/2) * powerof(b,p/2);
    
    if p % 2 == 0{
        return t;
    }
    return t * b;
}